### ISSUE TITLE
resource.ready(fn) doesn't register ready event

### ISSUE THREAD

**Issue by shirish87** [cayasso/primus-resource#1]
There's a bunch of commented code in `lib/client.js` since [this commit](https://github.com/cayasso/primus-resource/commit/26d9a1e18cc0b8dd91a3b85a38258b3a611ca663#diff-50cfa59973c04321b5da0c6da0fdf4feR156)
[And this](https://github.com/cayasso/primus-resource/blob/master/lib/client.js#L156) breaks all the sample code provided. :)

`resource.on('ready', fn)` works fine.

Thanks for this!

**Comment by cayasso** [comment]
Thanks for the heads up on this @shirish87 , I will add the corresponding fixes, I was really in a hurry and had to force the current release out, I will clean up this mess.

Thanks
JB

**Comment by shirish87** [comment]
no prb :)
Just a small unrelated question: Were you ever tempted to match the server-side method calls of the resources to the client-side (without requiring the on-prefix)?
I'm building something using this lib, and I kept forgetting prefixing resource methods with "on". Wrote a small wrapper to patch method names, so now it feels more RPC-like :)

**Comment by cayasso** [comment]
@shirish87, yes I know, I have think about that before but then I would need to keep track somehow on which events are public and which events are private, so I didn't want to jump into that issue and just leave it as simple as possible. But I am open for suggestions, the way it is was the best for my needs and my project requirement.

Here is the usual way I use it, I normally use a primary method and call this method from the on-prefix event, in my case I do it so because I have other modules that will also call the `talk` method like HTTP requests.

``` javascript
function Person(primus) {
  this.primus = primus;
  this.primus.resource(this.name);
}

Person.prototype.name = 'person';

Person.prototype.talk = function (message, fn) {
  console.log('Talking', message);
  fn(null, 'Just finish talking');
}

Person.prototype.ontalk = function (spark, message, fn) {
  this.talk(message, fn);
}
```

As I said the on-prefix was the best way I found to keep it simple and also avoid having to mantain a list or for instance having to define some kind of public/private flag on a method itself.

Thoughts?

Best,
JB

**Comment by cayasso** [comment]
Hey BTW I just released `2.0.1` with some refactoring and some cleanup, let me know if you have any question. I might build a more RPC like primus plugin but that would be not so soon :(

**Comment by shirish87** [comment]
Thanks so much for 2.0.1! This modules provides a nice base to build an RPC and even a 2-way RPC channel.

If it's about method visibility to the client, i guess it should be left to the user to hide those methods. Matching client and server-side method names is really intuitive when reading through code.
I have all my resources as classes in separate files/modules. module.exports exposes an instance of the class.
If there are methods that I'd like to hide, I follow some convention (prefix name with _) and at module.exports, I don't expose those methods.

`primus.resource('user', require('../resources/user').User)`

I have some code that iterates through all files in a `resources` folder and registers them as resources.

[Here's some of my code](https://gist.github.com/shirish87/73c4120f5fa0adbefd63#file-ws-js). It's part of a larger project so this won't work standalone. But you can see that I'm using primus-resource even for authentication :)
